src/server/service/debug.rs

```diff
@@ -143,10 +147,55 @@ impl debugpb_grpc::Debug for Service {
 
     fn scan_mvcc(
         &self,
-        _: RpcContext,
-        _: ScanMvccRequest,
-        _: ServerStreamingSink<ScanMvccResponse>,
+        ctx: RpcContext,
+        req: ScanMvccRequest,
+        sink: ServerStreamingSink<ScanMvccResponse>,
     ) {
-        unimplemented!()
+        let debugger = self.debugger.clone();
+        let deal_future = self.pool.spawn_fn(move || {
+            let mut dealer = MvccKVToSink(Some(sink));
+            let result = match debugger.scan_mvcc(req, &mut dealer) {
+                Ok(_) => Ok(dealer.0.unwrap()),
+                Err(e) => Err((dealer.0.unwrap(), e)),
+            };
+            future::result(result)
+        });
+
+        let finish_future = deal_future.then(|v| match v {
+            Ok(mut sink) => {
+                let f = future::poll_fn(move || sink.close())
+                    .map_err(|e| Service::on_grpc_error("scan_mvcc", e));
+                Box::new(f) as Box<Future<Item = (), Error = ()> + Send>
```

Comment:
I think spawning `f` here is fine.

Reply:
I tried but the `RpcContext` is not `Send`, so I can't call `ctx.spawn(f)` here.

Reply:
I mean call `ctx.spwan()` here, so `f` does not need to cast a `BoxFuture` .